### PR TITLE
show users more informative information on error

### LINKED ISSUE #15135: Link to a helpful FAQ in the error message "Oops, there was a problem during the request."
Often Matomo users who haven't yet setup Cron + Disable browser trigger archiving, index

It would be great to link from this message to the FAQ that explains this issue: https://matomo.org/faq/troubleshooting/faq_19489/

Maybe with append a text like `<a href='https://matomo.org/faq/troubleshooting/faq_19489/'>See the steps to resolve this issue in the FAQ</a>.`. 

Note:
* we don't want this extra text to be added on cloud. maybe we could link to the FAQ only when `enable_general_settings_admin=1` or `enable_plugins_admin=1`
* we will soon improve the FAQ at https://matomo.org/faq/troubleshooting/faq_19489/ and make it more clear / better

### PR DESCRIPTION
**Pull request by geekdenz**
fixes #15135

### Description:

Link to a helpful FAQ in the error message. This only shows when there is an error.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by geekdenz** [comment]
Are there more places this needs to be added?

**Comment by sgiehl** [comment]
I know this wasn't mentioned in the original issue, but wouldn't it makes sense to show this message only to super users? The message already says to contact the administrator if it happens again, which should be enough for normal users.

File: plugins/CoreHome/angularjs/widget-loader/widgetloader.directive.js [hunk 1]
```diff
@@ -35,6 +37,9 @@
 
                 return function (scope, element, attrs, ngModel) {
                     scope.widgetName = attrs.widgetName;
+                    scope.errorFaqLink = isGeneralSettingsAdminEnabled || isPluginsAdminEnabled ?
+                        _pk_translate('General_ErrorRequestFaqLink', ['<a target="_blank" href="https://matomo.org/faq/troubleshooting/faq_19489/">', '</a>'])
```
Comment (tsteur) [inline plugins/CoreHome/angularjs/widget-loader/widgetloader.directive.js]: fyi this needs a rel="noreferrer noopener" attribute for privacy and security

**Comment by mattab** [comment]
> Are there more places this needs to be added?

have you tried searching for the translation string like this: https://github.com/matomo-org/matomo/search?q=General_ErrorRequest where i can see these for example:

* https://github.com/matomo-org/matomo/blob/2b35b46a0988f8c3b6f848c59be8d7d918965a52/plugins/Morpheus/templates/ajaxMacros.twig#L20-L24
* https://github.com/matomo-org/matomo/blob/2b35b46a0988f8c3b6f848c59be8d7d918965a52/plugins/MultiSites/angularjs/dashboard/dashboard.directive.html#L61
* https://github.com/matomo-org/matomo/blob/b20dc4dc0e2c535e6a151e73a8631b1dc09d4821/plugins/Dashboard/javascripts/dashboardWidget.js#L176

Maybe we could add it there as well somehow?

> I know this wasn't mentioned in the original issue, but wouldn't it makes sense to show this message only to super users? The message already says to contact the administrator if it happens again, which should be enough for normal users.

Thanks for picking this up, yes, it would be important to only show the FAQ link to Super users :+1:

File: plugins/CoreHome/angularjs/widget-loader/widgetloader.directive.js (hunk 1 above)
Comment (geekdenz) [reply plugins/CoreHome/angularjs/widget-loader/widgetloader.directive.js]: OK, I added it. Good to know about this attribute and these values! I did not know these values and what they mean.

**Comment by geekdenz** [comment]
> [General_ErrorRequest](https://github.com/matomo-org/matomo/search?q=General_ErrorRequest)

Yes, I had found quite a few references myself, but thought I'd ask in the PR before committing too much effort into it.

File: core/View.php [hunk 2]
```diff
@@ -153,6 +153,11 @@ public function __construct($templateFile)
         $this->piwik_version = Version::VERSION;
         $this->userLogin = Piwik::getCurrentUserLogin();
         $this->isSuperUser = Access::getInstance()->hasSuperUserAccess();
+        // following is used in ajaxMacros called macro (showMoreHelp as passed in other templates) - requestErrorDiv
+        $this->isGeneralSettingsAdminEnabled = Controller::isGeneralSettingsAdminEnabled();
+        $this->isPluginsAdminEnabled = CorePluginsAdmin::isPluginsAdminEnabled();
```
Comment (sgiehl) [inline core/View.php]: Those variables aren't used in the view anymore, but only below. So guess they could be local variables only.

**Comment by geekdenz** [comment]
Hi @sgiehl . Do you have time to look at this, please?

**Review by sgiehl — APPROVED** [review]
Looks good to me now

**Comment by tassoman** [comment]
How do I can disable [this behaviour](https://github.com/matomo-org/matomo/blob/2b35b46a0988f8c3b6f848c59be8d7d918965a52/plugins/Morpheus/templates/ajaxMacros.twig#L20-L24)?  
It's ok by overwriting the template by a **ThemePlugin**?  
I don't want to bug administrators, but configured "From Recipient" installation global wide

**Comment by sgiehl** [comment]
I'm not sure what exactly you mean. The linked code is actually outdated. Meanwhile it's possible to configure a contact email address. See https://github.com/matomo-org/matomo/pull/17916

**Comment by tassoman** [comment]
Ok! Fine! I wasn't aware of the new configuration, was exactly what I was looking for.  
In the while, I was also able to override error message inside `ajaxMacros.twig` by putting an override into my ThemePlugin.
Thanks!

**Pull request merged**